redis.go

```diff
@@ -114,6 +122,26 @@ func (hs hooks) processTxPipeline(
 	return hs.processPipeline(ctx, cmds, fn)
 }
 
+func (hs hooks) withContext(ctx context.Context, fn func() error) error {
+	if ctx.Done() == nil {
+		return fn()
+	}
+
+	errc := make(chan error, 1)
+	go func() { errc <- fn() }()
+
+	select {
+	case <-ctx.Done():
+		err := ctx.Err()
+		if err == context.Canceled {
+			return err
+		}
```

Comment:
Is there a good reason not to return on all errors? If so - please add a comment.